javascript/javascript3/week3/js3w3-hw/js3w3-shoppingCart.js

```diff
@@ -0,0 +1,268 @@
+            msg.innerHTML = `provide product name to remove`
+
+        } else if (nameP.includes(product)) {
+            console.log(nameP);
+            const index = nameP.indexOf(product);
+            console.log(index);
+            if (index > -1) {
+                const removedItemFromArray = this.products.splice(index, 1);
+
+                console.log(removedItemFromArray);
+                const msg = document.querySelector('#msg');
+                msg.innerHTML = `${removedItemFromArray[0].name} deleted form list`
+                console.log(this.products);
+            }
+        } else {
+            const msg = document.querySelector('#msg');
+            msg.innerHTML = `${product} not available to delete`
+        }
+
+    }
+
+    searchProduct(productName) {
+        // const search = this.products.map((product) => product.name)
+        // if (search.includes(productName)) {
+        //     this.searchedProducts.push(productName)
+        //     console.log(this.searchedProducts);
+        //     return productName + " found";
+        // } else {
+        //     return productName + " Not found"
+        // }
+
+        const searchItems = this.products
+            .map((product) => product.name)
+            .filter((product) => product.includes(productName));
+        this.searchedProducts.push(productName)
+        const searchMsg = document.querySelector('#searchMsg')
+        searchMsg.innerHTML = searchItems
+
+        // console.log(this.searchedProducts);
+        console.log(searchItems);
+
+    }
+
+    getTotal() {
+        const total = this.products.map((product) => product.price).reduce(((total, currentPrice) => total + currentPrice), 0)
+        const getTotal = document.querySelector('#getTotal')
+        getTotal.innerHTML = ` Total Price: ${total} €`;
+  console.log(typeof total);
+        return total;
+    }
+
+    renderProducts() {
+        console.log(this.products);
+        const productCollection = document.querySelector('#productCollection');
+        const ProductName = document.querySelector('#ProductName');
+        const ProductPrice = document.querySelector('#ProductPrice');
+        const productUL = document.createElement('ul')
+        const prductli = document.createElement('li')
+
+        if (ProductName.value === '') {
+            const msg = document.querySelector('#addMsg');
+            msg.innerHTML = `provide product name to add`
+
+        } else {
+
+            prductli.innerHTML = `${ProductName.value} => Price: ${ProductPrice.value}`
+            productUL.appendChild(prductli);
+            productCollection.appendChild(productUL)
+            const msg = document.querySelector('#addMsg');
+            msg.innerHTML = `${ProductName.value} added to list`
+
+            console.log(ProductName.value + ProductPrice.value);
+        }
+
+
+    }
+
+    getUser(user) {
+        fetch(`https://jsonplaceholder.typicode.com/users/${user}`)
```

Comment:
it's better to handle error, when you fetch API

Reply:
ok, understood !